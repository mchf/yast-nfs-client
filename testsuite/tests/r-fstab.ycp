/**
 * Module:
 *   NFS client configuration
 *
 * Summary:
 *   Routines testuite
 *
 * Authors:
 *   Martin Vidner <mvidner@suse.cz>
 *
 * $Id$
 */
{
    // testedfiles: routines.ycp Testsuite.ycp

    include "testsuite.ycp";
    include "nfs/routines.ycp";

    string PathTooLong = "12345678901234567890123456789012345678901234567890123456789012345678901234567890";

    list<map> nfs_entries = [
	$[
	    "file":"/home",
	    "freq":0,
	    "mntops":"defaults",
	    "passno":0,
	    "spec":"foo.bar.com:/home",
	    "vfstype":"nfs4"
	 ],
	$[
	    "file":"/var/spool/mail",
	    "freq":0,
	    "mntops":"defaults",
	    "passno":0,
	    "spec":"foo.bar.com:/var/spool/mail",
	    "vfstype":"nfs"
	 ],
	$[
	    "file":"/install",
	    "freq":0,
	    "mntops":"hard,intr",
	    "passno":0,
	    // different from "file" (for order tests)
	    "spec":"foo.bar.com.tw:/local/install",
	    "vfstype":"nfs"
	]
	];

    DUMP ("FstabTableItems");
    TEST (``(FstabTableItems (nfs_entries)), [], nil);

    DUMP ("IsMpInFstab");
    // MountPoint in fstab
    TEST (``(IsMpInFstab (nfs_entries, "/home")), [], nil);
    // MountPoint in fstab
    TEST (``(IsMpInFstab (nfs_entries, "/install")), [], nil);
    // MountPoint NOT in fstab
    TEST (``(IsMpInFstab (nfs_entries, "/not/in/fstab")), [], nil);

    DUMP ("CheckPath");
    // Empty path
    TEST (``(CheckPath ("")), [], nil);
    // Path is too long (cca 80 chars)
    TEST (``(CheckPath (PathTooLong)), [], nil);
    // First slash is missing
    TEST (``(CheckPath ("not/begins/with/slash")), [], nil);
    // Too long with slash
    TEST (``(CheckPath ("/" + PathTooLong)), [], nil);
    // Regular path
    TEST (``(CheckPath ("/regular/path")), [], nil);
}
