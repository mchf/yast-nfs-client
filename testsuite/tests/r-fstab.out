Dump	FstabTableItems
Return	[`item (`id (0), "foo.bar.com ", "/home ", "/home ", "nfs4", "defaults "), `item (`id (1), "foo.bar.com ", "/var/spool/mail ", "/var/spool/mail ", "nfs", "defaults "), `item (`id (2), "foo.bar.com.tw ", "/local/install ", "/install ", "nfs", "hard,intr ")]
Dump	IsMpInFstab
Return	true
Return	true
Return	false
Dump	CheckPath
Return	false
Return	false
Return	false
Return	false
Return	true
